test/TestDataSpec.hs

```diff
@@ -14,6 +15,23 @@ import Data.Maybe
 errorMargin :: Float
 errorMargin = 1e-6
 
+assertBoidsListsEqual :: [Boid] -> [Boid] -> Assertion
+assertBoidsListsEqual expectedBoids actualBoids = sequence_ assertions
+    where expectedBoids' = sort expectedBoids
+          actualBoids' = sort actualBoids
+          lengthEquality = assertEqual "" (length expectedBoids) (length actualBoids)
+          assertions = lengthEquality : (map (uncurry $ assertBoidsEqual) $ zip expectedBoids' actualBoids')
+
+assertBoidsEqual :: Boid -> Boid -> Assertion
+assertBoidsEqual expectedBoid actualBoid = sequence_ assertions
+    where
+      properties = [ (fst . boidPosition, "X coordinates were not equal")
+                   , (snd . boidPosition, "Y coordinates were not equal")
+                   , (boidHeading, "Headings were not equal")
+                   , (boidSteer, "Steers were not equal") ]
+      assertions = map assertBoidsEqual properties
+      assertBoidsEqual (f, message) = assertApproxEqual message errorMargin (f expectedBoid) (f actualBoid)
```

Comment:
Please come up with another name for this function (the one in the `where` clause). It clashes with top-level `assertBoidsEqual`.